dashboard/config/routes.rb

```diff
@@ -91,6 +91,8 @@ module OPS
     end
   end
 
+  get '/dashboardapi/sections/membership', to: 'sections#membership'
```

Comment:
great question :-) yes, within a resource block, adding something to a `collection` block will create a route like /sections/membership, while adding something to a `member` block will create a route like /sections/:id/membership. The difference between `collection` and `member` is illustrated to some extent at http://guides.rubyonrails.org/routing.html.

Reply:
my first thought would be this (but i want to confirm since i don't see this anywhere else):

```
# Section API routes (JSON only)
concern :section_api_routes do
  resources :sections, only: [:index, :show, :create, :destroy] do
    resources :students, only: [:index, :update], controller: 'sections_students' do
      ...
  end

  get 'sections/membership', to: 'sections#membership'
end
```

Reply:
However the existing (linked above) collection block is under `sections/students`, so you'll need to add a new collection block under `sections` (sibling to the existing `member` block: https://github.com/code-dot-org/code-dot-org/blob/732c0b7e57f879fcc910d067098b886a54b0498b/dashboard/config/routes.rb#L85

Reply:
i never knew about `collection`! rails ❤️